js/pro/deribit.js

```diff
@@ -0,0 +1,843 @@
+'use strict';
+
+//  ---------------------------------------------------------------------------
+
+const deribitRest = require ('../deribit');
+const { NotSupported, ExchangeError } = require ('../base/errors');
+const { ArrayCache, ArrayCacheBySymbolById, ArrayCacheByTimestamp } = require ('./base/Cache');
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class deribit extends deribitRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchBalance': true,
+                'watchTicker': true,
+                'watchTickers': false,
+                'watchTrades': true,
+                'watchMyTrades': true,
+                'watchOrders': true,
+                'watchOrderBook': true,
+                'watchOHLCV': true,
+            },
+            'urls': {
+                'test': {
+                    'ws': 'wss://test.deribit.com/ws/api/v2',
+                },
+                'api': {
+                    'ws': 'wss://www.deribit.com/ws/api/v2',
+                },
+            },
+            'options': {
+                'timeframes': {
+                    '1m': 1,
+                    '3m': 3,
+                    '5m': 5,
+                    '15m': 15,
+                    '30m': 30,
+                    '1h': 60,
+                    '2h': 120,
+                    '4h': 180,
+                    '6h': 360,
+                    '12h': 720,
+                    '1d': '1D',
+                },
+                'currencies': [ 'BTC', 'ETH', 'SOL', 'USDC' ],
+            },
+            'streaming': {
+            },
+            'exceptions': {
+            },
+        });
+    }
+
+    requestId () {
+        const requestId = this.sum (this.safeInteger (this.options, 'requestId', 0), 1);
+        this.options['requestId'] = requestId;
+        return requestId;
+    }
+
+    async watchBalance (params = {}) {
+        /**
+         * @method
+         * @name deribit#watchBalance
+         * @see https://docs.deribit.com/#user-portfolio-currency
+         * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @param {object} params extra parameters specific to the deribit api endpoint
+         * @returns {object} a [balance structure]{@link https://docs.ccxt.com/en/latest/manual.html?#balance-structure}
+         */
+        await this.authenticate (params);
+        const messageHash = 'balance:BTC';
```

Comment:
if the user provides a different set of currencies, we might not even be watching the BTC balance 😅, can we change the hash to something more generic?